crates/biome_js_analyze/src/lint/nursery/use_readonly_class_properties.rs

```diff
@@ -286,51 +288,91 @@ declare_node_union! {
     pub AnyJsClassMethodBodyElement =
     JsArrowFunctionExpression |
     JsBlockStatement |
+    JsCallArguments |
     JsCallExpression |
     JsConditionalExpression |
+    JsConstructorClassMember |
     JsElseClause |
     JsExpressionStatement |
     JsFunctionBody |
+    JsGetterClassMember |
     JsGetterObjectMember |
     JsIfStatement |
     JsInitializerClause |
     JsMethodClassMember |
+    JsMethodObjectMember |
     JsObjectExpression |
     JsParenthesizedExpression |
     JsReturnStatement |
     JsSetterClassMember |
-    JsGetterClassMember |
     JsSetterObjectMember |
+    JsTemplateElement |
+    JsTemplateExpression |
     JsVariableDeclaration |
     JsVariableDeclarator |
     JsVariableStatement
 }
 
-enum MethodBodyElementOrStatementList<'a> {
-    MethodBodyElement(&'a AnyJsClassMethodBodyElement),
-    StatementList(&'a JsStatementList),
-    VariableDeclaratorList(&'a JsVariableDeclaratorList),
-    ObjectMemberList(&'a JsObjectMemberList),
+enum MethodBodyElementOrStatementList {
+    CallArgumentsList(JsCallArgumentList),
+    MethodBodyElement(AnyJsClassMethodBodyElement),
+    ObjectMemberList(JsObjectMemberList),
+    StatementList(JsStatementList),
+    TemplateElementList(JsTemplateElementList),
+    VariableDeclaratorList(JsVariableDeclaratorList),
 }
-
-impl<'a> MethodBodyElementOrStatementList<'a> {
+/// fn visit_fn_body_descendants will only visit the list of descendants listed here, more can be added if necessary
+impl MethodBodyElementOrStatementList {
     pub fn syntax(&self) -> &SyntaxNode<JsLanguage> {
         match self {
-            MethodBodyElementOrStatementList::MethodBodyElement(node) => node.syntax(),
-            MethodBodyElementOrStatementList::StatementList(list) => list.syntax(),
-            MethodBodyElementOrStatementList::VariableDeclaratorList(list) => list.syntax(),
-            MethodBodyElementOrStatementList::ObjectMemberList(list) => list.syntax(),
+            Self::CallArgumentsList(node) => node.syntax(),
+            Self::MethodBodyElement(node) => node.syntax(),
+            Self::ObjectMemberList(list) => list.syntax(),
+            Self::StatementList(list) => list.syntax(),
+            Self::TemplateElementList(list) => list.syntax(),
+            Self::VariableDeclaratorList(list) => list.syntax(),
         }
     }
 
+    pub fn from<T>(member: T) -> Self
```

Comment:
I think it’s an anti-pattern to implement this method outside the `From` trait. Best to create a separate `impl` block for it.